internal/hcsoci/credentials.go

```diff
@@ -0,0 +1,115 @@
+// +build windows
+
+package hcsoci
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/internal/log"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	"github.com/sirupsen/logrus"
+)
+
+// CCGState stores the id used when creating a ccg instance. Used when
+// closing a container to be able to release the instance.
+type CCGState struct {
+	// ID of container that instance belongs to.
+	id string
+}
+
+// Release calls into hcs to remove the ccg instance. These do not get cleaned up automatically
+// they MUST be explicitly removed with a call to ModifyServiceSettings. The instances will persist
+// unless vmcompute.exe exits or they are removed manually as done here.
+func (instance *CCGState) Release(ctx context.Context) error {
+	if err := removeCredentialGuard(ctx, instance.id); err != nil {
+		log.G(ctx).WithFields(logrus.Fields{
+			logrus.ErrorKey: err,
+			"containerID":   instance.id,
+		}).Error("failed to remove container credential guard instance")
+		return err
+	}
+	return nil
+}
+
+// CreateCredentialGuard creates a container credential guard instance and returns the state object to be placed in a v2 container doc.
```

Comment:
I would wrap this comment, the line is a bit long.

Reply:
FYI I wrap my comments at column 80.

Reply:
You can display a helpline line at column 80 in vscode by putting this in your config:

```
    "[go]": {
        "editor.rulers": [
            80
        ],
    },
```